crates/evm/src/execute.rs

```diff
@@ -1,17 +1,16 @@
 //! Traits for execution.
 
-use reth_execution_types::ExecutionOutcome;
-use reth_primitives::{BlockNumber, BlockWithSenders, Receipt, Request, U256};
-use reth_prune_types::PruneModes;
-use revm::db::BundleState;
-use revm_primitives::db::Database;
-use std::fmt::Display;
-
 #[cfg(not(feature = "std"))]
 use alloc::vec::Vec;
+use std::fmt::Display;
 
 pub use reth_execution_errors::{BlockExecutionError, BlockValidationError};
+use reth_execution_types::ExecutionOutcome;
+use reth_primitives::{BlockNumber, BlockWithSenders, Receipt, Request, U256};
+use reth_prune_types::PruneModes;
 pub use reth_storage_errors::provider::ProviderError;
+use revm::db::BundleState;
+use revm_primitives::db::Database;
```

Comment:
as suggested, would have been good if an issue was opened to discuss this before the work was put in to impl this. we don't want this lint because it doesn't make a difference between `pub use` and private `use` statements.

if you can revert the mixing of `pub use` and private `use` statements, and remove the lint from ci, we can merge this.

Reply:
I'm not sure this is possible here:
- From what I see in [group_imports](https://rust-lang.github.io/rustfmt/?version=master&search=import#group_imports), `StdExternalCrate` generates three groups but does not differentiate `pub use` and `use`
- The thing you are mentioning here seems to be related to https://github.com/rust-lang/rustfmt/issues/5083#issuecomment-2036247567 but unfortunately I don't think this is possible right now, wdyt?

Reply:
Don't hesitate to close if needed this is not a problem, this was just adding one line to the fmt file and running `cargo +nightly fmt` that's why I didn't open an issue for this before implementation.

Reply:
you can fix it manually?

Reply:
I don't think so, I tried but as soon as we provide a `group_import` element in the fmt, it seems that all imports are automatically rearranged at save...